### ISSUE TITLE
No Audio Output on some devices

### ISSUE THREAD

**Issue by melliiii** [Melvin-Abraham/Google-Assistant-Unofficial-Desktop-Client#495]
**Describe the bug**
Hey, for some odd reason I dont get a Audio Output (TTS) on my Headset, but on my Monitor Speakers. The Feedback "Ping" works on every device.

**To Reproduce**
1. Set Headset as Default Audio Device
2. Tell Assistant to do something
3. See the Audio on Windows Volume Mixer, but no Output can be heard
4. Test that Headphones work with other programs (In my case Spotify and Youtube in Firefox)
4. Change Monitor Speakers as Default Audio Device
5. Tell Assistant to do something
6. Now I get Audio via my Monitor

**Expected behavior**
No matter what audio device is set as default, the TTS audio should play back.

**Desktop (please complete the following information):**
 - OS: Windows 10 Pro Build 19042.1165
 - Version 1.0.0-rc.1 for Windows
 - Headphones: Logitech G533
 - Monitor: AOC 2460G5
 
 I tried restarting the computer etc. but no luck. Also seems like I'm the only one with the problem
 
 EDIT: Clarification that I get the Feedback, but not the TTS

**Comment by melliiii** [comment]
"Enable Audio Output" Option in the Settings is enabled.

**Comment by im-coder-lg** [comment]
Try setting the headphones for the speaker in the settings.

**Comment by melliiii** [comment]
![grafik](https://user-images.githubusercontent.com/46799789/130718537-fcd0afa7-4412-4aca-8592-1da480598e2a.png)
Only Headphones are enabled and it wont output TTS. If i only enable the Monitor (directly below) it outputs

**Comment by im-coder-lg** [comment]
Not there, in GAUDC

**Comment by melliiii** [comment]
![grafik](https://user-images.githubusercontent.com/46799789/130718780-edae0835-f461-40e0-b457-1abb4236e21d.png)
![grafik](https://user-images.githubusercontent.com/46799789/130718805-3e7f4c64-7b23-4665-ac63-5e1649173916.png)
 
Thats all the settings i got

**Comment by im-coder-lg** [comment]
Wait what? Lemme check for a minute

**Comment by im-coder-lg** [comment]
I am posting an image, check that

**Comment by im-coder-lg** [comment]
![image](https://user-images.githubusercontent.com/76845820/130719143-dc06dd99-9031-4a0f-908a-0b658c40c86b.png)
See that :arrow_up:

**Comment by im-coder-lg** [comment]
If it is not there, you have to **build** it yourself. Tell me if you need here.

**Comment by melliiii** [comment]
Dont have those options :o Do i have build it from source or is there a compiled version?

**Comment by im-coder-lg** [comment]
Building from source is best. Do you know how?

**Comment by im-coder-lg** [comment]
If you don't then refer this: https://github.com/Melvin-Abraham/Google-Assistant-Unofficial-Desktop-Client/issues/490#issuecomment-901915422

**Comment by melliiii** [comment]
Yes I do ^^ Ill get back to you when im done

**Comment by im-coder-lg** [comment]
Don't forget: you need:

- Git
- Node JS LTS
- Windows Build Tools(`npm install windows-build-tools`? am I right here?)
- Visual Studio 2015/2019
- Python 3.8(Anaconda3 is suggested because it has the best Python ever.)
I think that's it.

**Comment by melliiii** [comment]
I wanna build on my Programming VM, but it takes a while to boot xD I got all of that :3

**Comment by im-coder-lg** [comment]
Good!

**Comment by im-coder-lg** [comment]
What os is it?

**Comment by melliiii** [comment]
Ubuntu :D

**Comment by im-coder-lg** [comment]
That surely does work.

**Comment by melliiii** [comment]
Yeah, all that stuff is on a Server thats in my bedroom ^^ But its rather old and the raid module only supports HDDs so it takes a while to start the Hyper-V instance etc

**Comment by im-coder-lg** [comment]
You must always remember, if there are new commits/features you ever want, you have to build it until the next release with the compiled version is out. And the next release will be a little sooner.

**Comment by im-coder-lg** [comment]
Now, does it successfully work?

**Comment by melliiii** [comment]
NPM install takes a while, ill get back to you when its done

**Comment by im-coder-lg** [comment]
> * OS: Windows 10 Pro Build 19042.1165
> * Version 1.0.0-rc.1 for Windows
> * Headphones: Logitech G533
> * Monitor: AOC 2460G5

Oof, the version:
> * Version 1.0.0-rc.1 for Windows

Try using the latest release candidate(1.0.0-rc2) or build from source(which you are doing)

**Comment by melliiii** [comment]
Ywah thats a type, it was rc.2

**Comment by im-coder-lg** [comment]
okay, now if the build succeeds, mainly, go check the dist folder. I don't know *how* the folder is but it might show something like `linux-unpacked` or something like that.

**Comment by melliiii** [comment]
Okay something is reeealllyyy weird

**Comment by melliiii** [comment]
Same issue as before, but now i have some more settings

**Comment by im-coder-lg** [comment]
what? change the speaker thing there. its not weird, it's full of features.

**Comment by melliiii** [comment]
If i start OBS to record, OBS captures the Audio i cant hear and if i replay the recording i really hear the sound. OBS only captures what i should hear so something fishy is going on

**Comment by im-coder-lg** [comment]
check this out:
![image](https://user-images.githubusercontent.com/76845820/130719143-dc06dd99-9031-4a0f-908a-0b658c40c86b.png)

change the speaker thing

**Comment by melliiii** [comment]
I did, changing works, but i still cant hear the TTS Audio but i see it in Windows Volume Mixer. And if i record what i should hear in OBS (desktop audio) i also hear what GAUDC says in the recording

**Comment by im-coder-lg** [comment]
> If i start OBS to record, OBS captures the Audio i cant hear and if i replay the recording i really hear the sound. OBS only captures what i should hear so something fishy is going on

that is OBS. it captures all the audio, and I know it because, I was seeing a chess video and I wanted to test OBS but the audio went really down. But is the build and changing the speaker inside GAUDC work?

**Comment by melliiii** [comment]
Yeah changing speakers works... Maybe i just have to live with the fact, that i dont get Audio Playback of the text-to-speech voice over my headset and just let it output on my monitor

**Comment by im-coder-lg** [comment]
strange... /ping @Melvin-Abraham is this really fine? I have a *wired* headphone and I hear the ping in that properly. Try using wired headphones.

**Comment by im-coder-lg** [comment]
> Yeah changing speakers works... Maybe i just have to live with the fact, that i dont get Audio Playback of the text-to-speech voice over my headset and just let it output on my monitor

wait does it work

**Comment by melliiii** [comment]
Just to clarify: No matter what audio device i use, the "ping" always is hearable. Text To Speech only works on the Monitor, not on the headset.

**Comment by im-coder-lg** [comment]
brb one min...

**Comment by im-coder-lg** [comment]
it works on my PC, I hear both the ping and the audio response. Maybe you must use wired headphones.

**Comment by melliiii** [comment]
Hmm i think i just let the output go to the monitor... Thats fine for now, but thank you very much for your help!

**Comment by im-coder-lg** [comment]
Anytime!

**Comment by Melvin-Abraham** [comment]
@melliiii This is definitely a bug. I will try to investigate.
_Reopening issue_

**Comment by im-coder-lg** [comment]
oh yeah @Melvin-Abraham it doesn't work even if he switches speaker output in the settings, can you check about that? He told that he let the output go to the monitor too.

**Comment by melliiii** [comment]
> 
> 
> @melliiii This is definitely a bug. I will try to investigate.
> _Reopening issue_

If you need any logs/specifications let me know ^^



> 
> 
> oh yeah @Melvin-Abraham it doesn't work even if he switches speaker output in the settings, can you check about that? He told that he let the output go to the monitor too.

She*, actually :D But you couldnt have known

**Comment by im-coder-lg** [comment]
Sorry and oh yeah there was a log screenshot in this issue somewhere. Just FYI.

**Comment by Melvin-Abraham** [comment]
@melliiii What options do you have for **"Microphone Source"** and **"Speaker Source"** when connecting to your Bluetooth headset (connect first and then open the settings) and which options did you choose for respective the settings?

**Comment by melliiii** [comment]
Microphone Options:
![grafik](https://user-images.githubusercontent.com/46799789/130774505-458a3081-712f-4ed2-b6e2-943cce917e11.png)

Speaker Options:
![grafik](https://user-images.githubusercontent.com/46799789/130774576-d7004f39-e67f-4d6f-b3c7-90a6be69c820.png)

Selected: Default Anua Mic CM 900 and currently 2460G5 (would be replaced by "Default Logitech G533" if I would get Audio on that thing)

**Comment by im-coder-lg** [comment]
Wait what? What was that? It is supposed to work! I use a Windows 7 computer with wired headphones and it works successfully. Ping+Text-to-Speech works(both) but what is that? is that buggy?

**Comment by melliiii** [comment]
Hm maybe its a Windows 10 thing?

**Comment by melliiii** [comment]
![grafik](https://user-images.githubusercontent.com/46799789/130778825-38ea423e-5914-4c01-833a-2b52c431f9a3.png)
When i switch between Headset and monitor and it plays audio i get this Exception that isnt handled. But the Audio plays fine as long as I selected the monitor

**Comment by melliiii** [comment]
I fixed it by just using Voicemeeter. Still seems like im the only one with the problem and i cant really troubleshoot whats wrong because i dont really know JS, electron and the other frameworks youre using 😅

**Comment by Melvin-Abraham** [comment]
@melliiii Try using "Default" or "Lautsprecher" for Speaker Source

**Comment by melliiii** [comment]
Using the "Lautsprecher" Variant also doesnt work

**Comment by Melvin-Abraham** [comment]
I tested it using my wireless headset and it works just fine with "Headset" set for both microphone and speaker source... Can you try with "Default"?

**Comment by im-coder-lg** [comment]
Did the same thing without changing(might have, idk) for my wired headphones and it works great too. Maybe, try everything related to your headphones(every name it shows for the headphones)
